pyqtgraph/parametertree/parameterTypes/basetypes.py

```diff
@@ -229,6 +229,12 @@ def optsChanged(self, param, opts):
 
         if 'readonly' in opts:
             self.updateDefaultBtn()
+
+            if opts['readonly']:
+                self.displayLabel.setTextInteractionFlags(QtCore.Qt.TextSelectableByMouse)
+            else:
+                self.displayLabel.setTextInteractionFlags(QtCore.Qt.LinksAccessibleByMouse)
```

Comment:
For PyQt6 support you need to specify the full enum namespace, so it should be 

```python
            if opts['readonly']:
                self.displayLabel.setTextInteractionFlags(QtCore.Qt.TextInteractionFlag.TextSelectableByMouse)
            else:
                self.displayLabel.setTextInteractionFlags(QtCore.Qt.TextInteractionFlag.LinksAccessibleByMouse)
```

Luckily the full name enum namespace works for all the other bindings we support too 👍🏻